### ISSUE TITLE
🐛 Sub-optimal computation of exact mapper, heuristic performs better than exact mapper

### ISSUE THREAD

**Issue by jeanlyc** [cda-tum/qmap#251]
### mqt.qmap version

mqt-qmap 2.1.1 via pip

### OS

Ubuntu

### Description

See MEW below:
```
from qiskit import QuantumCircuit
from mqt import qmap

circ = QuantumCircuit(4)
circ.cx(1,0)
circ.cx(0,1)
circ.cx(2,1)
circ.cx(1,2)
circ.cx(3,2)

arch = qmap.Architecture(
    5,
    {
        (1, 0),
        (2, 0),
        (2, 1),
        (4, 2),
        (3, 2),
        (3, 4),
    },
)

qc_mapped_exact, res = qmap.compile(circ, arch, method="exact", post_mapping_optimizations=False)
qc_mapped_heu, res = qmap.compile(circ, arch, method="heuristic", post_mapping_optimizations=False)

print(f"Exact: {qc_mapped_exact.count_ops()}") # gives 12 addtional H - sub-optimal but correct
print(f"Heuristic: {qc_mapped_heu.count_ops()}") # gives 8 addtional H - correct + optimal
```

### Expected behavior

- I would assume that the heuristic approach, in contrast to the exact approach, is not better in terms of added number of gates
- The solution of the exact mapper is not the minimal number of addtional H gates

### How to Reproduce

- Run MWE above
- MWE can be computed via pen&paper

**Comment by burgholzer** [comment]
Thanks for bringing this up and taking the time to create the minimal reproducer. That certainly looks like a bug and I'll look into it. I'll come back to you once I found something.

**Comment by burgholzer** [comment]
Just released 2.1.2 with a fix for the issue above. Should be on PyPI later today.
Thanks again for bringing this up and coming up with such a small example. Made it fairly easy to dig to the ground of the issue.
